src/movies.js

```diff
@@ -1,25 +1,151 @@
+		(movie) => movie.director === "Steven Spielberg" && movie.genre.includes("Drama")
+	);
+	return stevenMovies.length;
+}
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+	if (moviesArray.length === 0) {
+		return 0;
+	} else {
+		const totalScore =
+			moviesArray.reduce((acc, movie) => {
+				if (!movie.score) {
+					return acc;
+				} else {
+					return acc + movie.score;
+				}
+			}, 0) / moviesArray.length;
+		return Math.round(totalScore * 100) / 100;
+	}
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+	const dramaMovies = moviesArray.filter((movie) => movie.genre.includes("Drama"));
+	return scoresAverage(dramaMovies);
+}
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+function orderByYear(moviesArray) {
+	const newMoviesArray = [...moviesArray];
+	newMoviesArray.sort((a, b) => {
+		if (a.year < b.year) {
+			return -1;
+		} else if (a.year > b.year) {
+			return 1;
+		} else if (a.title < b.title) {
+			return -1;
+		} else if (a.title > b.title) {
+			return 1;
+		} else {
+			return 0;
+		}
+	});
+	return newMoviesArray;
+}
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically(moviesArray) {}
+function orderAlphabetically(moviesArray) {
+	const newMoviesArray = [...moviesArray];
+	newMoviesArray.sort((a, b) => {
+		if (a.title < b.title) {
+			return -1;
+		} else if (a.title > b.title) {
+			return 1;
+		} else {
+			return 0;
+		}
+	});
+	const newMoviesArrayTitle = newMoviesArray.map((movie) => {
+		return movie.title;
+	});
+	const firstTwentyMovies = newMoviesArrayTitle.slice(0, 20);
+	return firstTwentyMovies;
+}
 
 // BONUS - Iteration 7: Time Format - Turn duration of the movies from hours to minutes
-function turnHoursToMinutes(moviesArray) {}
+function turnHoursToMinutes(moviesArray) {
+	const minutedMovies = JSON.parse(JSON.stringify(moviesArray));
+
+	minutedMovies.map((movie) => {
+		let totalMinutes = 0;
+		let splitsArray = [];
+
+		if (movie.duration.includes("h")) {
+			if (movie.duration.includes("min")) {
+				splitsArray = movie.duration.split("min");
+				splitsArray = splitsArray[0].split("h ");
+				totalMinutes = splitsArray[0] * 60 + parseInt(splitsArray[1]);
+			} else {
+				splitsArray = movie.duration.split("h");
+				totalMinutes = splitsArray[0] * 60;
+				console.log("");
+			}
+		} else {
+			splitsArray = movie.duration.split("min");
+			totalMinutes = parseInt(splitsArray[0]);
+		}
+
+		movie.duration = totalMinutes;
+		return movie;
+	});
+	return minutedMovies;
+}
 
 // BONUS - Iteration 8: Best yearly score average - Best yearly score average
-function bestYearAvg(moviesArray) {}
+function bestYearAvg(moviesArray) {
```

Comment:
Te pongo por aqui otra posible solución:
function bestYearAvg(moviesArray) {

  if (moviesArray.length === 0) return null;
  const years = moviesArray.map((el) => el.year).sort();
  let bestYear = 0;
  let bestAverage = 0;
  for (const year of years) {
    const thisYear = moviesArray.filter((el) => el.year === year);
    console.log(year);
    const average = scoresAverage(thisYear);
    console.log(average, bestAverage);
    if (average > bestAverage) {
      bestYear = year;
      bestAverage = average;
    }
  }
  return `The best year was ${bestYear} with an average score of ${bestAverage}`;
}